Getting the variables from arbitrary SPARQL patterns (OpTable)

Getting variables from arbitrary SPARQL patterns (such as `OpTable`) is marked as a TODO in `se.liu.ida.hefquin.jenaext.sparql.algebra.OpUtils`. This causes queries such as the one below to fail with an exception:

```sparql
SELECT * WHERE {
  SERVICE <https://query.wikidata.org/sparql> { ?y a <http://nothing> }
  SERVICE <http://dbpedia.org/sparql>         { OPTIONAL { ?y a <http://nothing> } }
}
```

```java
Attention: The query result may be incomplete because the following exception was caught when executing the query plan

1 java.lang.UnsupportedOperationException: Getting the variables from arbitrary SPARQL patterns is an open TODO (type of Jena Op in the current case: org.apache.jena.sparql.algebra.op.OpTable).
StackTrace:
java.lang.Exception: Exception occurred when outputting the result of a SELECT query using the Jena machinery.
        at se.liu.ida.hefquin.engine.HeFQUINEngine._execSelectQuery(HeFQUINEngine.java:304)
        at se.liu.ida.hefquin.engine.HeFQUINEngine._execAndPrint(HeFQUINEngine.java:257)
        at se.liu.ida.hefquin.engine.HeFQUINEngine.executeQueryAndPrintResult(HeFQUINEngine.java:172)
        at se.liu.ida.hefquin.cli.RunQueryWithoutSrcSel.exec(RunQueryWithoutSrcSel.java:140)
        at org.apache.jena.cmd.CmdMain.mainMethod(CmdMain.java:86)
        at org.apache.jena.cmd.CmdMain.mainRun(CmdMain.java:55)
        at org.apache.jena.cmd.CmdMain.mainRun(CmdMain.java:43)
        at se.liu.ida.hefquin.cli.RunQueryWithoutSrcSel.main(RunQueryWithoutSrcSel.java:56)
Caused by: java.lang.UnsupportedOperationException: Getting the variables from arbitrary SPARQL patterns is an open TODO (type of Jena Op in the current case: org.apache.jena.sparql.algebra.op.OpTable).
        at se.liu.ida.hefquin.jenaext.sparql.algebra.OpUtils.addVariablesFromPattern(OpUtils.java:67)
        at se.liu.ida.hefquin.jenaext.sparql.algebra.OpUtils.addVariablesFromPattern(OpUtils.java:77)
        at se.liu.ida.hefquin.jenaext.sparql.algebra.OpUtils.addVariablesFromPattern(OpUtils.java:54)
        at se.liu.ida.hefquin.jenaext.sparql.algebra.OpUtils.getVariablesInPattern(OpUtils.java:36)
        at se.liu.ida.hefquin.base.query.impl.GenericSPARQLGraphPatternImpl2.getAllMentionedVariables(GenericSPARQLGraphPatternImpl2.java:124)
        at se.liu.ida.hefquin.engine.queryplan.executable.impl.ops.BaseForExecOpSequentialBindJoinSPARQL.<init>(BaseForExecOpSequentialBindJoinSPARQL.java:25)
        at se.liu.ida.hefquin.engine.queryplan.executable.impl.ops.ExecOpSequentialBindJoinSPARQLwithVALUESorFILTER.<init>(ExecOpSequentialBindJoinSPARQLwithVALUESorFILTER.java:85)
        at se.liu.ida.hefquin.engine.queryplan.physical.impl.PhysicalOpBindJoinSPARQL.createExecOp(PhysicalOpBindJoinSPARQL.java:193)
        at se.liu.ida.hefquin.engine.queryplan.physical.impl.PhysicalOpBindJoinSPARQL.createExecOp(PhysicalOpBindJoinSPARQL.java:127)
        at se.liu.ida.hefquin.engine.queryproc.impl.compiler.QueryPlanCompilerForPushBasedExecution$Worker._createThreads(QueryPlanCompilerForPushBasedExecution.java:113)
        at se.liu.ida.hefquin.engine.queryproc.impl.compiler.QueryPlanCompilerForPushBasedExecution$Worker.createThreads(QueryPlanCompilerForPushBasedExecution.java:80)
        at se.liu.ida.hefquin.engine.queryproc.impl.compiler.QueryPlanCompilerForPushBasedExecution.createThreads(QueryPlanCompilerForPushBasedExecution.java:46)
        at se.liu.ida.hefquin.engine.queryproc.impl.compiler.QueryPlanCompilerForPushBasedExecution.compile(QueryPlanCompilerForPushBasedExecution.java:39)
        at se.liu.ida.hefquin.engine.queryproc.impl.QueryProcessorImpl.processQuery(QueryProcessorImpl.java:80)
        at se.liu.ida.hefquin.jenaintegration.sparql.engine.main.OpExecutorHeFQUIN$MainQueryIterator.nextStage(OpExecutorHeFQUIN.java:171)
        at org.apache.jena.sparql.engine.iterator.QueryIterRepeatApply.makeNextStage(QueryIterRepeatApply.java:100)
        at org.apache.jena.sparql.engine.iterator.QueryIterRepeatApply.hasNextBinding(QueryIterRepeatApply.java:60)
        at org.apache.jena.sparql.engine.iterator.QueryIteratorBase.hasNext(QueryIteratorBase.java:122)
        at org.apache.jena.sparql.engine.iterator.QueryIteratorWrapper.hasNextBinding(QueryIteratorWrapper.java:37)
        at org.apache.jena.sparql.engine.iterator.QueryIteratorBase.hasNext(QueryIteratorBase.java:122)
        at org.apache.jena.sparql.engine.iterator.QueryIteratorWrapper.hasNextBinding(QueryIteratorWrapper.java:37)
        at org.apache.jena.sparql.engine.iterator.QueryIteratorBase.hasNext(QueryIteratorBase.java:122)
        at java.base/java.util.Iterator.forEachRemaining(Iterator.java:132)
        at org.apache.jena.sparql.exec.RowSetStream.forEachRemaining(RowSetStream.java:79)
        at org.apache.jena.sparql.exec.RowSetMem.<init>(RowSetMem.java:74)
        at org.apache.jena.sparql.exec.RowSetMem.create(RowSetMem.java:44)
        at org.apache.jena.sparql.exec.RowSetStream.rewindable(RowSetStream.java:64)
        at org.apache.jena.riot.rowset.rw.RowSetWriterText.output$(RowSetWriterText.java:130)
        at org.apache.jena.riot.rowset.rw.RowSetWriterText.output(RowSetWriterText.java:118)
        at org.apache.jena.riot.rowset.rw.RowSetWriterText.output(RowSetWriterText.java:114)
        at org.apache.jena.riot.rowset.rw.RowSetWriterText.write(RowSetWriterText.java:59)
        at org.apache.jena.sparql.resultset.ResultsWriter.write(ResultsWriter.java:186)
        at org.apache.jena.sparql.resultset.ResultsWriter.write(ResultsWriter.java:156)
        at org.apache.jena.sparql.resultset.ResultsWriter.write(ResultsWriter.java:149)
        at org.apache.jena.sparql.util.QueryExecUtils.outputResultSet(QueryExecUtils.java:158)
        at se.liu.ida.hefquin.engine.HeFQUINEngine._execSelectQuery(HeFQUINEngine.java:301)
        ... 7 more
```

--- a/hefquin-base/src/main/java/se/liu/ida/hefquin/jenaext/sparql/algebra/OpUtils.java
+++ b/hefquin-base/src/main/java/se/liu/ida/hefquin/jenaext/sparql/algebra/OpUtils.java
@@ -12,6 +12,7 @@ import org.apache.jena.sparql.algebra.op.OpFilter;
 import org.apache.jena.sparql.algebra.op.OpJoin;
 import org.apache.jena.sparql.algebra.op.OpLeftJoin;
 import org.apache.jena.sparql.algebra.op.OpService;
+import org.apache.jena.sparql.algebra.op.OpTable;
 import org.apache.jena.sparql.algebra.op.OpUnion;
 import org.apache.jena.sparql.core.Var;
 import org.apache.jena.sparql.core.Vars;
@@ -62,6 +63,9 @@ public class OpUtils
 		else if ( op instanceof OpExtend opExtend ){
 			addVariablesFromPattern( acc, opExtend.getSubOp() );
 			acc.addAll( opExtend.getVarExprList().getVars() );
+		}
+		else if ( op instanceof OpTable opTable ){
+			acc.addAll( opTable.getTable().getVars() );
 		}
 		else {
 			throw new UnsupportedOperationException("Getting the variables from arbitrary SPARQL patterns is an open TODO (type of Jena Op in the current case: " + op.getClass().getName() + ").");

